crates/sui-framework/sources/crypto/ecdsa_k1.move

```diff
@@ -3,22 +3,27 @@
 
 module sui::ecdsa_k1 {
 
-    // TODO document this
+    /// Error if the public key cannot be recovered from the signature. 
     const EFailToRecoverPubKey: u64 = 0;
-    const EInvalidSignature: u64 = 1;
 
+    /// Error if the signature is invalid.
+    const EInvalidSignature: u64 = 1;
+    
+    /// Hash function name that are valid for ecrecover and secp256k1_verify. 
+    const KECCAK256: u8 = 3;
```

Comment:
We should have a common shared list somewhere and document why KECCAK256: u8 = 3; sha256  = 4 etc (btw it doesn't matter if they collide with error codes (ie starting from zero), these flags serve a different role). Can we share these flags between k1 and r1 (instead of rewriting them?)

Reply:
reason why I had them here is bc they are accessible for ppl using these contract to easily find the definition. i can put this in hash.move, and link them in ecdsa_k1 and ecdsa_r1 module?